doc/website/blog/2024-05-24-atlas-gorm-provider-view.mdx

```diff
@@ -0,0 +1,284 @@
+ a consistent interface to users, abstracting the underlying data structures.
+
+> Making liberal use of views is a key aspect of good SQL database design.
+>
+> — <cite>**Postgres documentation**</cite>
+
+## The current landscape
+
+Despite the benefits, views are often underutilized in many projects. Defining and managing database views can be tedious
+ and error-prone, especially when it comes to migration. 
+
+GORM is a fantastic ORM library that supports auto creation for database schema 
+with the [AutoMigrate](https://gorm.io/docs/migration.html#Auto-Migration) feature. AutoMigrate does 
+its job, but it's only sufficient in simple cases. When you need to 
+fully control and adapt [versioned migrations](/concepts/declarative-vs-versioned#versioned-migrations), but 
+still want to use your favorite GORM, Atlas is the answer.
+
+## Introducing Atlas + GORM
+
+Atlas can automatically plan database schema migrations for developers using GORM, 
+with the help of the [GORM Atlas Provider](https://github.com/ariga/atlas-provider-gorm). Moreover, not only tables but 
+also views are now fully supported. Unlike GORM itself, with Atlas, views can be managed and versioned in the same way as tables, as they should be.
+
+## Demo Time!
+
+Let's walk through a step-by-step example of using GORM Atlas Provider
+to automatically plan schema migrations for tables and views in a GORM project.
+
+### Installation
+
+If you haven't already, install Atlas from macOS or Linux by running:
+
+```bash
+curl -sSf https://atlasgo.sh | sh
+```
+
+See [atlasgo.io](https://atlasgo.io/getting-started#installation) for more installation options.
+
+In addition, the view feature is only available for logged-in users, run the following command to login:
+
+```bash
+atlas login
+```
+
+Install the provider by running:
+
+```bash
+go get -u ariga.io/atlas-provider-gorm
+```
+
+#### Step 1: Create a GORM Application
+
+Models are defined using normal structs. For views, we define a struct and 
+implement the [ViewDefiner]([https://pkg.go.dev/ariga.io/atlas-provider-gorm/gormschema#ViewDefiner]) interface. The `ViewDef(dialect string)` method receives the 
+`dialect` argument to determine the SQL dialect to generate the view. It is helpful for generating the view definition for different SQL dialects if needed.
+
+Let's create a file that will contain our database models. We will call it `models/models.go`
+
+```go title="models/models.go"
+package models
+
+import (
+  "ariga.io/atlas-provider-gorm/gormschema"
+  "gorm.io/gorm"
+)
+
+// User is a regular gorm.Model stored in the "users" table.
+type User struct {
+  gorm.Model
+  Name   string
+  Age    int
+  Gender string
+}  
+
+// WorkingAgedUser is mapped to the VIEW definition below.
+type WorkingAgedUser struct {
+  Name string
+  Age  int
+}
+
+func (WorkingAgedUser) ViewDef(dialect string) []gormschema.ViewOption {
+    return []gormschema.ViewOption{
+        gormschema.CreateStmt(`
```

Comment:
This is the core part of the blogpost where you reveal the new tech. 

Technical blog readers are skeptical and curious. 

Show off the different alternatives - using GORM syntax (first. this is the familiar API to the user), then alternatively show providing the query. Mention that it is DB specific, and show how to use a switch case statement where necessary.

Reply:
> Mention that it is DB specific, and show how to use a switch case statement where necessary.

Hey @rotemtam, I need a little help with this part. Should I mention it as the main example or keep my original example and include the switch case as an additional section?

The reason is that I want to show the code differences later, so I use multiple lines for the SQL statement. Including the switch case in the example will make it extremely verbose

Reply:
You can use a `<Tabs>` component to make examples more condensed, if that works